lib/gollum-lib/committer.rb

```diff
@@ -75,14 +75,17 @@ def parents
     # path   - The String path to be added
     # data   - The String wiki data to store in the tree map.
     #
-    # Raises Gollum::DuplicatePageError if a matching filename already exists.
+    # Raises Gollum::DuplicatePageError if a matching filename already exists, unless force_overwrite is explicitly enabled.
     # This way, pages are not inadvertently overwritten.
     #
     # Returns nothing (modifies the Index in place).
-    def add_to_index(path, data, options = {})
+    def add_to_index(path, data, options = {}, force_overwrite = false)
       if index.current_tree
-        tree = @wiki.page_file_dir ? index.current_tree / @wiki.page_file_dir : index.current_tree
-        raise DuplicatePageError.new(path) if !page_path_scheduled_for_deletion?(index.tree, path) && tree.blobs.find {|blob| blob.name == path}
+        tree = index.current_tree
+        unless force_overwrite
```

Comment:
Maybe: `unless force_overwrite || page_path_scheduled_for_deletion?(index.tree, path)`

If it's a delete operation, there is no need to call `tree / path`.

Reply:
Good suggestion, will implement.